### PR TITLE
Update service-catalog chart (remove apiserver)

### PR DESCRIPTION
**Pull request by adamwalach**
**Description**

Changes proposed in this pull request:

- removed api-server based SC chart
- updated grafana dashboards
- removed configuration parameters (version selection)

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/5524

### CODE REVIEW

File: resources/service-catalog/charts/catalog/values.yaml [hunk 1]
```diff
@@ -1,134 +1,60 @@
 # Default values for Service Catalog
 # service-catalog image to use
-image: quay.io/kubernetes-service-catalog/service-catalog:v0.1.43
+#TODO: change that!
+image: awalach/service-catalog:crds
```
Comment (jasiu001) [inline resources/service-catalog/charts/catalog/values.yaml]: "change that" when?

Comment (adamwalach) [reply resources/service-catalog/charts/catalog/values.yaml]: When we'll have new SC release. Possibly before merging this PR :)

**Comment by jasiu001** [comment]
SC works, metrics are fine, LGTM

File: resources/service-catalog/charts/catalog/Chart.yaml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 name: catalog
-description: service-catalog API server and controller-manager helm chart
-version: 0.1.40
+description: service-catalog webhook server and controller-manager helm chart
+version: 0.1.41
```
Comment (mszostok) [inline resources/service-catalog/charts/catalog/Chart.yaml]: we need to adjust that after releasing ServiceCatalog

File: resources/service-catalog/charts/catalog/templates/controller-manager-deployment.yaml [hunk 3]
```diff
@@ -7,24 +7,17 @@ metadata:
...
-  # Strategy overwrites the default service-catalog strategy because of using helm '--wait' flag.
-  # As a result, Tiller waits for the readiness of all of the components, and then evaluates the readiness.
-  strategy:
-    type: RollingUpdate
-    rollingUpdate:
-      maxUnavailable: 0
   template:
     metadata:
       annotations:
-        sidecar.istio.io/inject: "true"
```
Comment (mszostok) [inline resources/service-catalog/charts/catalog/templates/controller-manager-deployment.yaml]: add that annotation under: `.Values.controllerManager.annotations`

File: resources/service-catalog/charts/catalog/templates/metrics.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 apiVersion: authentication.istio.io/v1alpha1
 kind: Policy
 metadata:
-  name: {{ template "fullname" . }}-controller-manager
```
Comment (mszostok) [inline resources/service-catalog/charts/catalog/templates/metrics.yaml]: why we changed that?

File: resources/service-catalog/charts/catalog/values.yaml (hunk 1 above)
Comment (adamwalach) [reply resources/service-catalog/charts/catalog/values.yaml]: Changed

**Pull request merged**